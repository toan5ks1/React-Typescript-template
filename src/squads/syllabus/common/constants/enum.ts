export enum MIMETypes {
    CSV = "text/csv",
    VIDEO = "video/*",
    IMAGE = "image/*",
    AUDIO = "audio/*",
    PDF = "application/pdf",
    MSWORD = "application/msword",
}
