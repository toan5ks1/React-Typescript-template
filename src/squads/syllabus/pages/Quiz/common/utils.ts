import { Answer, Quiz } from "src/squads/syllabus/models/quiz";

import { removeAllLoadingImage } from "src/squads/syllabus/pages/Quiz/components/WYSWYG/wyswyg-utils";

export const removeAllLoadingImageFromHTML = () => {
    const loadingImages = document.querySelectorAll(`[data-testid="Editor__loading-image"]`);
    if (loadingImages && loadingImages.length > 0) {
        loadingImages.forEach((loadingImage) => {
            loadingImage.remove();
        });
    }
};

export const removeLoadingImageFromQuizPayload = (quiz: Quiz) => {
    const answerList = new Map();
    quiz.answer.list.forEach((answer) => {
        answerList.set(answer.id, {
            ...answer,
            content: removeAllLoadingImage(answer.content),
        });
    });
    removeAllLoadingImageFromHTML();
    return {
        ...quiz,
        answer: {
            ...quiz.answer,
            list: answerList,
        },
        question: {
            ...quiz.question,
            content: removeAllLoadingImage(quiz.question.content),
        },
        explanation: {
            ...quiz.explanation,
            content: removeAllLoadingImage(quiz.explanation.content),
        },
    };
};

// Check answer or alternative by index in array (return by getAllAnswerGroupsByGroupKey)
export const isMainAnswerFIB = (index: number) => {
    return index === 0;
};

// Push answer (with alternatives) in array by groupKey
// All these array will need to render UI of FIB question
export const getAllAnswerGroupsByGroupKey = (answers: Answer[]): Array<Answer[]> => {
    const quantity = answers.length;
    const map: Map<string, Answer[]> = new Map();

    for (let i = 0; i < quantity; i++) {
        const currentAnswer = answers[i];
        const { groupKey } = currentAnswer;

        if (groupKey) {
            const mapValues = map.get(groupKey);

            if (mapValues) {
                map.set(groupKey, [...mapValues, currentAnswer]);
                continue;
            }

            map.set(groupKey, [currentAnswer]);
        }
    }
    return Array.from(map.values());
};
