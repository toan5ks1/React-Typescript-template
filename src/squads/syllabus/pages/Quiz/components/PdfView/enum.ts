export enum SelectFieldState {
    RECT_TYPE = "RECT_TYPE",
    FIELD_TYPE = "FIELD_TYPE",
}
