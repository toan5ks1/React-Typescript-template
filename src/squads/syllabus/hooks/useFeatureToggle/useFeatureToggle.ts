import { useEffect, useState } from "react";

import { Features } from "src/common/constants/enum";

import useFeatureController from "src/squads/syllabus/hooks/useFeatureController";

const useFeatureToggle = (toggleName: Features, shouldSubscribe: boolean = true) => {
    const { featureController: featureControllerUnleash } = useFeatureController();

    const [isEnabled, setIsEnabled] = useState(
        featureControllerUnleash.isFeatureEnabled(toggleName)
    );

    useEffect(() => {
        if (!shouldSubscribe) return;

        featureControllerUnleash.subscribeToRemoteChanges(() => {
            setIsEnabled(featureControllerUnleash.isFeatureEnabled(toggleName));
        });
    }, [featureControllerUnleash, shouldSubscribe, toggleName]);

    return { isEnabled };
};

export default useFeatureToggle;
