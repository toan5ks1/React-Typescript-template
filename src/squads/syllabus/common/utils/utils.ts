import { Entities, EurekaEntities } from "src/common/constants/enum";
import { DynamicFormFieldValue } from "src/squads/syllabus/common/types/common";

import {
    isTaskAssignment,
    isLO,
    LOAndAssignmentType,
} from "src/squads/syllabus/pages/Book/components/LOAndAssignment/models";

export const convertToString = (value?: DynamicFormFieldValue): string => {
    return typeof value === "string" ? value : "";
};

function stringifyReplacer(_key: string | number | symbol, value: any) {
    if (typeof value === "undefined") {
        return "undefined";
    }

    const specialTypes = ["function", "bigint", "symbol"];
    // We use different operation for special kinds
    if (specialTypes.includes(typeof value)) {
        return value.toString();
    }

    return value;
}

export function safeStringify(target: any, spacing?: number): string {
    return JSON.stringify(target, stringifyReplacer, spacing);
}

export function getResource(item: LOAndAssignmentType) {
    if (isLO(item)) return Entities.LOS;
    if (isTaskAssignment(item.type as string)) return EurekaEntities.TASK_ASSIGNMENTS;
    return EurekaEntities.ASSIGNMENTS;
}
