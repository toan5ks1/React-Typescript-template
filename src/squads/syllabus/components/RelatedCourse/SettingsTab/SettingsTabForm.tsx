import { ChangeEvent, useCallback } from "react";

import { Entities, NotifyTypes, Features } from "src/common/constants/enum";
import { ArrayElement, LocationManyType } from "src/common/constants/types";
import { arrayHasItem } from "src/common/utils/other";
import { Lesson_CoursesOneQuery } from "src/squads/syllabus/services/bob/bob-types";
import inferMutation from "src/squads/syllabus/services/infer-mutation";

import { Box } from "@mui/material";
import DividerDashed from "src/components/Divider/DividerDashed";
import TypographyWithValue from "src/components/Typographys/TypographyWithValue";
import AvatarInput from "src/squads/syllabus/components/InputFiles/AvatarInput";

import useFeatureToggle from "src/squads/syllabus/hooks/useFeatureToggle";
import useResourceTranslate from "src/squads/syllabus/hooks/useResourceTranslate";
import useShowSnackbar from "src/squads/syllabus/hooks/useShowSnackbar";
import useTranslate from "src/squads/syllabus/hooks/useTranslate";

export interface SettingsTabProps {
    course: ArrayElement<Lesson_CoursesOneQuery["courses"]>;
    locations: LocationManyType[];
    isLoadingLocation: boolean;
    refetch: () => void;
}

export interface SettingsTabFormProps extends SettingsTabProps {}

export const getLocationsList = (locations: LocationManyType[]): string => {
    if (!arrayHasItem(locations)) return "";
    return locations.map((location) => location.name).join(", ");
};

const SettingsTabForm = (props: SettingsTabFormProps) => {
    const { course, locations, isLoadingLocation, refetch } = props;

    const showSnackbar = useShowSnackbar();
    const t = useTranslate();
    const tCourse = useResourceTranslate(Entities.COURSES);

    const { mutate: submitForm, isLoading } = inferMutation({
        entity: "coursesYasuo",
        action: "courseCreate",
    })({
        onSuccess: () => {
            showSnackbar(tCourse("updateCourseIconSuccess"));
            refetch();
        },
        onError: () => {
            showSnackbar(t("ra.common.updatedFail"), NotifyTypes.ERROR);
        },
    });

    const { isEnabled: enabledTeachingMethod } = useFeatureToggle(
        Features.LESSON_COURSE_BACK_OFFICE_TEACHING_METHOD
    );

    const { isEnabled: enabledLocationField } = useFeatureToggle(
        Features.LESSON_COURSE_DETAIL_BACK_OFFICE_LOCATION_FIELD
    );
    const onSubmitAvatarInput = useCallback(
        (e: ChangeEvent<HTMLInputElement>) => {
            if (Boolean(e.target.files?.length)) {
                const listFile = Array.from(e.target.files || []);
                return submitForm({
                    name: course.name,
                    course_id: course.course_id,
                    school_id: course.school_id,
                    display_order: course.display_order,
                    icon: course.icon,
                    chapter_ids: [],
                    files: listFile,
                });
            }
        },
        [submitForm, course]
    );

    const onRemove = useCallback(() => {
        return submitForm({
            name: course.name,
            course_id: course.course_id,
            school_id: course.school_id,
            display_order: course.display_order,
            chapter_ids: [],
            icon: undefined,
        });
    }, [submitForm, course]);

    const locationNames = getLocationsList(locations);

    return (
        <Box>
            <Box display="flex" justifyContent="center" pb={3}>
                <AvatarInput
                    name="icon"
                    shouldDelete
                    src={course.icon || undefined}
                    readOnly={isLoading}
                    onChange={onSubmitAvatarInput}
                    onRemove={onRemove}
                />
            </Box>
            <DividerDashed />

            <Box pt={3}>
                <TypographyWithValue
                    label={tCourse("courseName")}
                    value={course.name}
                    variant="horizontal"
                    xsLabel={2}
                    xsValue={10}
                    dataTestidValue="SettingsTabForm__courseName"
                />
                {enabledLocationField && (
                    <Box pt={1.5}>
                        <TypographyWithValue
                            label={tCourse("location")}
                            value={locationNames}
                            variant="horizontal"
                            xsLabel={2}
                            xsValue={10}
                            dataTestidValue="SettingsTabForm__locationsName"
                            isLoading={isLoadingLocation}
                        />
                    </Box>
                )}
                {enabledTeachingMethod && (
                    <Box pt={1.5}>
                        <TypographyWithValue
                            label={tCourse("teachingMethod")}
                            value={course.teaching_method ? tCourse(course.teaching_method) : ""}
                            variant="horizontal"
                            xsLabel={2}
                            xsValue={10}
                            dataTestidValue="SettingsTabForm__teachingMethod"
                        />
                    </Box>
                )}
            </Box>
        </Box>
    );
};

export default SettingsTabForm;
