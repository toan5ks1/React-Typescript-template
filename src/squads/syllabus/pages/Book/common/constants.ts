import { convertEnumKeys } from "src/squads/syllabus/common/utils/enum";

import { BookType } from "manabuf/common/v1/contents_pb";

export const BookTypeKeys = convertEnumKeys(BookType);
