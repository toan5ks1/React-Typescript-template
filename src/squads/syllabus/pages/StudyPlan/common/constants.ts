import { convertEnumKeys } from "src/squads/syllabus/common/utils/enum";

import { StudyPlanItemStatus } from "manabuf/eureka/v1/assignments_pb";
import { StudyPlanStatus } from "manabuf/eureka/v1/enums_pb";

import { FormFilterStudyPlanValues, StudyPlanFormData } from "./types";

export const datePattern = /^(\d{4}\/\d{2}\/\d{2})$/;
export const dateTimePattern = /^(\d{4}\/\d{2}\/\d{2},\s\d{2}:\d{2})$/;

export const StudyPlanStatusKey = convertEnumKeys(StudyPlanStatus);
export const StudyPlanItemStatusKey = convertEnumKeys(StudyPlanItemStatus);

export const studyPlanFilterDefaultValues: FormFilterStudyPlanValues = {
    grades: [],
    books: [],
    archived: false,
};

export const defaultStudyPlanFormValues: StudyPlanFormData = {
    name: "",
    book: null,
    grades: [],
    trackSchoolProgress: false,
};
