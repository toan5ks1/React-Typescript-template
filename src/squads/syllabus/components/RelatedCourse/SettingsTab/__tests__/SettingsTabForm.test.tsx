import { PropsWithChildren } from "react";

import { Features } from "src/common/constants/enum";
import { convertToString } from "src/squads/syllabus/common/utils/utils";
import inferMutation from "src/squads/syllabus/services/infer-mutation";

import SettingsTabForm, { getLocationsList, SettingsTabFormProps } from "../SettingsTabForm";

import { fireEvent, render, RenderResult, screen, waitFor, within } from "@testing-library/react";
import useFeatureToggle from "src/squads/syllabus/hooks/useFeatureToggle";
import TestApp from "src/squads/syllabus/test-utils/TestApp";
import TestHookFormProvider from "src/squads/syllabus/test-utils/TestHookFormProvider";

jest.mock("src/squads/syllabus/hooks/useShowSnackbar", () => ({
    __esModule: true,
    default: () => jest.fn(),
}));

const mockUploadAvatar = jest.fn();

jest.mock("src/squads/syllabus/services/infer-mutation", () => {
    return {
        __esModule: true,
        default: jest.fn(),
    };
});

jest.mock("src/squads/syllabus/hooks/useFeatureToggle", () => ({
    __esModule: true,
    default: jest.fn(),
}));

const defaultProps: SettingsTabFormProps = {
    course: {
        course_books: [],
        name: "Course 01",
        course_id: "abc",
        icon: "",
        school_id: 1,
    },
    isLoadingLocation: false,
    locations: [
        { location_id: "location_id_1", name: "Center 1" },
        { location_id: "location_id_2", name: "Center 2" },
    ],
    refetch: jest.fn(),
};

const renderComponent = (enableTeachingMethod: boolean = true, enableLocation: boolean = true) => {
    (inferMutation as jest.Mock).mockImplementation(
        (_inferredBase: { entity: "coursesYasuo"; action: "courseCreate" }) => {
            return () => ({ mutate: mockUploadAvatar });
        }
    );
    (useFeatureToggle as jest.Mock).mockImplementation((feature: Features) => {
        let enable: boolean = false;
        switch (feature) {
            case Features.LESSON_COURSE_BACK_OFFICE_TEACHING_METHOD:
                enable = enableTeachingMethod;
                break;
            case Features.LESSON_COURSE_DETAIL_BACK_OFFICE_LOCATION_FIELD:
                enable = enableLocation;
                break;
            default:
                break;
        }
        return {
            isEnabled: enable,
        };
    });

    return render(<SettingsTabForm {...defaultProps} />, {
        wrapper: ({ children }: PropsWithChildren<{}>) => (
            <TestApp>
                <TestHookFormProvider>{children}</TestHookFormProvider>
            </TestApp>
        ),
    });
};

describe("<SettingsTabForm/> in course enabledTeachingMethod true", () => {
    let wrapper: RenderResult;
    beforeEach(() => {
        wrapper = renderComponent();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should show correctly course name", () => {
        const courseName = screen.getByTestId("SettingsTabForm__courseName");
        expect(within(courseName).getByText(defaultProps.course.name)).toBeTruthy();
    });

    it("should call mockUploadAvatar function when upload avatar for course", async () => {
        expect(screen.getByTestId("AvatarInput__labelHtmlFor")).toBeInTheDocument();
        const uploader = screen.getByTestId("AvatarInput__input");
        const file = new File(["(⌐□_□)"], "avatar.png", { type: "image/png" });
        expect(uploader).toBeInTheDocument();

        fireEvent.change(uploader, {
            target: { files: [file] },
        });

        await waitFor(() =>
            expect(screen.getByTestId("AvatarInput__root").style.backgroundImage).not.toEqual(
                "undefined"
            )
        );
        expect(mockUploadAvatar).toHaveBeenCalled();
    });

    it("should not call mockUploadAvatar function when upload empty file", async () => {
        expect(screen.getByTestId("AvatarInput__labelHtmlFor")).toBeInTheDocument();
        const uploader = screen.getByTestId("AvatarInput__input");
        expect(uploader).toBeInTheDocument();

        fireEvent.change(uploader, { target: { files: [] } });

        await waitFor(() =>
            expect(screen.getByTestId("AvatarInput__root").style.backgroundImage).toEqual("none")
        );
        expect(mockUploadAvatar).not.toHaveBeenCalled();
    });

    it("should show locations name correctly", () => {
        const locationNames = getLocationsList(defaultProps.locations);
        expect(screen.getByText(convertToString(locationNames))).toBeInTheDocument();
    });

    it("should show teaching method field", () => {
        expect(screen.getByTestId("SettingsTabForm__teachingMethod")).toBeInTheDocument();
    });
});

describe("<SettingsTabForm/> in course with unleash flag", () => {
    it("should render correctly with teaching method unleash key false", () => {
        renderComponent(false, true);
        expect(screen.queryByTestId("SettingsTabForm__teachingMethod")).not.toBeInTheDocument();
    });

    it("should render correctly with location unleash key false", () => {
        renderComponent(true, false);
        expect(screen.queryByTestId("SettingsTabForm__locationsName")).not.toBeInTheDocument();
    });
});
