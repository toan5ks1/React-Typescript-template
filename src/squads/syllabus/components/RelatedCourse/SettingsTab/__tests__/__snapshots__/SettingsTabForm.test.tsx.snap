// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsTabForm/> in course enabledTeachingMethod true should match snapshot 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-bottom: 24px;
}

.emotion-2 {
  position: relative;
  border-radius: 50%;
  height: 112px;
  width: 112px;
  background-color: #EEEEEE;
}

.emotion-3 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -moz-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 2.1875rem;
  color: #757575;
}

.emotion-5 {
  display: none;
}

.emotion-6 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: #E0E0E0;
  border-bottom-width: thin;
  border-top: 1px dashed #E0E0E0;
  border-bottom: 0px solid;
  height: 0;
  background-color: transparent;
}

.emotion-7 {
  padding-top: 24px;
}

.emotion-8 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  word-break: break-word;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 16.666667%;
  -ms-flex-preferred-size: 16.666667%;
  flex-basis: 16.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 16.666667%;
}

@media (min-width:600px) {
  .emotion-9 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:960px) {
  .emotion-9 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:1440px) {
  .emotion-9 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:1920px) {
  .emotion-9 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

.emotion-10 {
  margin: 0;
  font-size: 0.75rem;
  font-weight: 400;
  letter-spacing: 0.4px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.66;
  color: #757575;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 83.333333%;
  -ms-flex-preferred-size: 83.333333%;
  flex-basis: 83.333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 83.333333%;
}

@media (min-width:600px) {
  .emotion-11 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:960px) {
  .emotion-11 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:1440px) {
  .emotion-11 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:1920px) {
  .emotion-11 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-13 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

.emotion-14 {
  padding-top: 12px;
}

<div>
  <form
    novalidate=""
  >
    <div
      class="MuiBox-root emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <div
          class="MuiBox-root emotion-2"
          data-testid="AvatarInput__root"
          style="background-image: none; background-size: cover;"
        >
          <label
            data-testid="AvatarInput__labelHtmlFor"
            for="icon"
            style="position: absolute; width: 100%; height: 100%; cursor: pointer;"
          >
            <span
              class="MuiBox-root emotion-3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge emotion-4"
                data-testid="AddAPhotoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6h-3.17L16 4h-6v2h5.12l1.83 2H21v12H5v-9H3v9c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zM8 14c0 2.76 2.24 5 5 5s5-2.24 5-5-2.24-5-5-5-5 2.24-5 5zm5-3c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3zM5 6h3V4H5V1H3v3H0v2h3v3h2z"
                />
              </svg>
            </span>
          </label>
          <div
            class="MuiBox-root emotion-5"
          >
            <input
              accept="image/*"
              data-testid="AvatarInput__input"
              id="icon"
              type="file"
            />
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth emotion-6"
      />
      <div
        class="MuiBox-root emotion-7"
      >
        <div
          class="MuiGrid-root MuiGrid-container emotion-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 emotion-9"
          >
            <span
              class="MuiTypography-root MuiTypography-caption emotion-10"
            >
              Course Name
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 emotion-11"
            data-testid="SettingsTabForm__courseName"
          >
            <div
              class="MuiBox-root emotion-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 emotion-13"
              >
                Course 01
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-14"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-8"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 emotion-9"
            >
              <span
                class="MuiTypography-root MuiTypography-caption emotion-10"
              >
                Location
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 emotion-11"
              data-testid="SettingsTabForm__locationsName"
            >
              <div
                class="MuiBox-root emotion-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 emotion-13"
                >
                  Center 1, Center 2
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-14"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-8"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 emotion-9"
            >
              <span
                class="MuiTypography-root MuiTypography-caption emotion-10"
              >
                Teaching Method
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 emotion-11"
              data-testid="SettingsTabForm__teachingMethod"
            >
              <div
                class="MuiBox-root emotion-12"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 emotion-13"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <input
      data-testid="TextFieldHF__submit"
      type="submit"
    />
  </form>
</div>
`;
